test-framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -110,8 +119,18 @@ public Transport original() {
      * Adds a rule that will cause every send request to fail, and each new connect since the rule
      * is added to fail as well.
      */
-    public void addFailToSendNoConnectRule(DiscoveryNode node) {
-        addDelegate(node, new DelegateTransport(original) {
+    public void addFailToSendNoConnectRule(TransportService transportService) {
+        for (TransportAddress transportAddress : transportService.boundAddress().boundAddresses()) {
+            addFailToSendNoConnectRule(transportAddress);
+        }
+    }
+
+    /**
+     * Adds a rule that will cause every send request to fail, and each new connect since the rule
+     * is added to fail as well.
+     */
+    public void addFailToSendNoConnectRule(TransportAddress transportAddress) {
```

Comment:
can we document transportAddress, how we extract addresses from it and how it will affect the sendRequest command (which uses a disco node)?